python-avd/pyavd/_cv/client/studio_topology.py

```diff
@@ -0,0 +1,359 @@
+
+            request.values.append(
+                InputsConfig(
+                    key=InputsKey(
+                        studio_id=TOPOLOGY_STUDIO_ID,
+                        workspace_id=workspace_id,
+                        path=RepeatedString(values=["devices", str(device_index), "inputs", "device"]),
+                    ),
+                    inputs=json.dumps(device_info),
+                ),
+            )
+
+        index_offset = len(studio_inputs.get("devices", []))
+        # Add any devices not part of the topology studio already.
+        for index, device in enumerate(device_inputs_by_id.items()):
+            device_id, device_inputs = device
+            device_index = index + index_offset
+            device_entry = {
+                "inputs": {"device": {**device_inputs, "modelName": "", "interfaces": []}},
+                "tags": {"query": f"device:{device_id}"},
+            }
+            request.values.append(
+                InputsConfig(
+                    key=InputsKey(
+                        studio_id=TOPOLOGY_STUDIO_ID,
+                        workspace_id=workspace_id,
+                        path=RepeatedString(values=["devices", str(device_index)]),
+                    ),
+                    inputs=json.dumps(device_entry),
+                ),
+            )
+
+        client = InputsConfigServiceStub(self._channel)
+        responses = client.set_some(request, metadata=self._metadata, timeout=timeout + len(request.values) * 0.1)
+        return [response.key async for response in responses]
+
+    @GRPCRequestHandler()
+    async def set_topology_decommissions(
+        self: CVClientProtocol,
+        device_ids: list[str],
+        workspace_id: str,
+        timeout: float = DEFAULT_API_TIMEOUT,
+    ) -> list[DecommissionConfig]:
+        """
+        Decommission devices using arista.studio_topology.v1.DecommissionConfigService.SetSome API.
+
+        Parameters:
+            device_ids: Device IDs to decommission.
+            workspace_id: Workspace ID to decommission the devices from.
+            timeout: Timeout in seconds.
+
+        Returns:
+            List of DecommissionConfig objects.
+        """
+        decommission_configs = [DecommissionConfig(key=DeviceKey(device_id=device_id, workspace_id=workspace_id)) for device_id in device_ids]
+        request = DecommissionConfigSetSomeRequest(decommission_configs)
+        client = DecommissionConfigServiceStub(self._channel)
+        responses = client.set_some(request, metadata=self._metadata, timeout=timeout)
+
+        return [response.key async for response in responses]
+
+    @GRPCRequestHandler()
+    async def get_all_decommissions(
+        self: CVClientProtocol,
+        device_ids: list[str],
+        workspace_id: str,
+        state: Literal["success", "unspecified", "failure"],
+        timeout: float = DEFAULT_API_TIMEOUT,
+    ) -> list[DecommissionConfig]:
```

Comment:
The code is returning a single `Decommission` object. Is that expected?
```suggestion
    ) -> Decommission:
```